Bottom: 205d0e16f0a477e8ee02c60c0e10fd52d8c6dc35
Top:    a27f9216cafd665edbfe0bd9a074521b8f0d3949
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-02-24 00:04:11 +0000

Use strdup to copy environment strings into our settings structure.

getenv returns a const char * which is part of the environment and must
not be free'd.


---

diff --git a/src/main.c b/src/main.c
index 400aebc..bb79991 100644
--- a/src/main.c
+++ b/src/main.c
@@ -563,6 +563,8 @@ REGISTER_ATTRIBUTES (main_attribute_list)
  */
      static void settings_post_process ()
 {
+  char *tmps;
+
   if (Settings.LibraryCommand[0] != PCB_DIR_SEPARATOR_C && Settings.LibraryCommand[0] != '.')
     {
       Settings.LibraryCommand
@@ -602,14 +604,18 @@ REGISTER_ATTRIBUTES (main_attribute_list)
    * to call
    */
   if (Settings.MakeProgram == NULL) {
-    Settings.MakeProgram = getenv ("PCB_MAKE_PROGRAM");
+    tmps = getenv ("PCB_MAKE_PROGRAM");
+    if (tmps != NULL)
+      Settings.MakeProgram = strdup (tmps);
   }
   if (Settings.MakeProgram == NULL) {
     Settings.MakeProgram = strdup ("make");
   }
 
   if (Settings.GnetlistProgram == NULL) {
-    Settings.GnetlistProgram = getenv ("PCB_GNETLIST");
+    tmps = getenv ("PCB_GNETLIST");
+    if (tmps != NULL)
+      Settings.GnetlistProgram = strdup (tmps);
   }
   if (Settings.GnetlistProgram == NULL) {
     Settings.GnetlistProgram = strdup ("gnetlist");
